### PR TITLE
[bitcoin-move][breaking]Refactor ordinals inscription updater

### PR DESCRIPTION
**Pull request by jolestar**
## Summary

1. Add `copy` and `drop` to SimpleMap and SimpleMultiMap.
2. Follow the ordinals inscription_updater to implement inscription_updater in Move.
3. Add id(InscriptionID), location(SatPoint), charms(u16) fields in Inscription. cc @popcnt1 @yubing744 
4. Add unbound_inscription_count and lost_sats in InscriptionStore. cc @popcnt1 

part of #2402 and #2403

### CODE REVIEW

File: frameworks/bitcoin-move/sources/inscription_updater.move [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+            }
+        };
+        
+        let input_idx = 0;
+        let input_len = vector::length(txinput);
+        while(input_idx < input_len){
+            //let input = vector::borrow(txinput, input_idx);
+            let utxo = vector::borrow_mut(input_utxos, input_idx);
+            if (is_coinbase){
+                total_input_value = total_input_value + ord::subsidy_by_height(block_height);
```
Comment (baichuan3) [inline frameworks/bitcoin-move/sources/inscription_updater.move:172]: 如果是 coinbase 交易，应该在累加 total_input_value后continue，继续循环执行，对应的 utxo为 null utxo

File: frameworks/bitcoin-move/sources/inscription_updater.move [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+
+                let curse = ord::handle_curse_inscription(&envelope);
+
+                let flotsam = Flotsam{
+                    inscription_id: inscription_id,
+                    offset: offset,
+                    new: option::some(FlotsamNew{
+                        cursed: option::is_some(&curse) && !jubilant,
+                        fee: 0,
+                        //TODO should we handle the hidden
```
Comment (baichuan3) [inline frameworks/bitcoin-move/sources/inscription_updater.move:239]: hidden 字段和home_inscription没有特别看明白，不确定对offset的影响

File: frameworks/bitcoin-move/sources/inscription_updater.move [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+
+        let output_idx = 0;
+        let output_len = vector::length(txoutput);
+        let flotsam_idx = 0;
+        let flotsam_len = vector::length(&floating_inscriptions);
+        while(output_idx < output_len){
+            let output = vector::borrow(txoutput, output_idx);
+            let value = types::txout_value(output);
+            let output_script_buf = types::txout_script_pubkey(output);
+            let is_op_return = script_buf::is_op_return(output_script_buf);
```
Comment (baichuan3) [inline frameworks/bitcoin-move/sources/inscription_updater.move:285]: ordinal原始实现里没有处理 op_return

File: frameworks/bitcoin-move/sources/inscription_updater.move [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+            let output_address = types::txout_object_address(output);
+            let end = output_value + value;
+            
+            while(flotsam_idx < flotsam_len){
+                let flotsam = vector::borrow(&floating_inscriptions, flotsam_idx);
+                if (flotsam.offset >= end){
+                    break
+                };
+
+                let new_satpoint = ord::new_satpoint(types::new_outpoint(txid, (output_idx as u32)), flotsam.offset - output_value);
```
Comment (baichuan3) [inline frameworks/bitcoin-move/sources/inscription_updater.move:295]: new_satpoint的计算比较复杂，看了几遍，总觉得跟 ordinal 实现有差异，需要测试用例覆盖

File: frameworks/bitcoin-move/sources/inscription_updater.move (hunk 3 above)
Comment (jolestar) [reply frameworks/bitcoin-move/sources/inscription_updater.move:285]: 最新的代码里有了。

File: frameworks/bitcoin-move/sources/inscription_updater.move (hunk 1 above)
Comment (jolestar) [reply frameworks/bitcoin-move/sources/inscription_updater.move:155]: 修复了

File: crates/rooch/src/commands/statedb/commands/inscription.rs [hunk 5]
```diff
@@ -71,14 +72,20 @@ impl InscriptionSource {
...
+            outpoint: outpoint.into(),
             offset: src.satpoint_offset,
+        };
+        Inscription {
+            id,
+            location,
             sequence_number: src.sequence_number,
             inscription_number: src.inscription_number.unsigned_abs(),
             is_curse: src.inscription_number.is_negative(),
+            //TODO how to get charms
```
Comment (popcnt1) [inline crates/rooch/src/commands/statedb/commands/inscription.rs:87]: I could export charms from ordinal index

File: crates/rooch/src/commands/statedb/commands/inscription.rs [hunk 6]
```diff
@@ -136,6 +143,9 @@ pub(crate) fn create_genesis_inscription_store_object(
     let inscription_store = InscriptionStore {
         cursed_inscription_count,
         blessed_inscription_count,
+        //TODO set unbound_inscription_count and lost_sats
+        unbound_inscription_count: 0,
+        lost_sats: 0,
```
Comment (popcnt1) [inline crates/rooch/src/commands/statedb/commands/inscription.rs:148]: For statedb/genesis: count lost_sats by charm flag

**Review by popcnt1 — APPROVED** [review]

File: crates/rooch/src/commands/statedb/commands/inscription.rs (hunk 6 above)
Comment (jolestar) [reply crates/rooch/src/commands/statedb/commands/inscription.rs:148]: The ordinals has a statistic_to_count db to get the counts [UpgradeCap](https://github.com/ordinals/ord/blob/75bf04b22107155f8f8ab6c77f6eefa8117d9ace/src/index/updater.rs#L604-L626)

**Pull request merged**